src/stores/RegionOverviewData.js

```diff
@@ -0,0 +1,55 @@
+import {observable, asMap, action} from 'mobx';
+import {without} from 'lodash';
+
+export default class RegionOverviewData {
+  @observable dataByRegion = asMap({})
+
+  @action
+  initializeRegionData(regionName, period, serviceName = 'aggregated') {
+    if (!this.dataByRegion.get(regionName)) {
+      this.dataByRegion.set(regionName, asMap({}));
+    }
+    if (!this.dataByRegion.get(regionName).get(period)) {
+      this.dataByRegion.get(regionName).set(period, asMap({}));
+    }
+    if (!this.dataByRegion.get(regionName).get(period).get(serviceName)) {
+      this.dataByRegion.get(regionName).get(period).set(serviceName, observable({
+        sla: null,
+        availability: null,
+        health: null,
+        performance: null
+      }));
+    }
+    return this.dataByRegion.get(regionName).get(period).get(serviceName);
+  }
+
+  @action
+  update(regionName, period, serviceName = 'aggregated', plainOverviewData) {
+    const {
+      sla,
+      availability,
+      health,
+      performance
+    } = plainOverviewData;
+    Object.assign(this.initializeRegionData(regionName, period, serviceName), {
```

Comment:
You can write in the following way:

Object.assign(this.initializeRegionData(...), {...plainOverviewData, lastUpdate: new Date()});